flutter/lib/screens/home_screen.dart

```diff
@@ -0,0 +1,75 @@
+//Flutter Packages
+import 'dart:async';
+import 'package:flutter/material.dart';
+//Third party packages
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:lottie/lottie.dart';
+import 'package:simple_connection_checker/simple_connection_checker.dart';
+//My Packages
+import '../screens/offline_screen.dart';
+import '../screens/qr_code_screen.dart';
+import '../screens/scanner_screen.dart';
+import '../widgets/google_fab/action_button.dart';
+import '../widgets/google_fab/expandable_fab.dart';
+
+class HomeScreen extends StatefulWidget {
+
+  static const routeName = "/home_screen";
+
+  const HomeScreen({Key? key}) : super(key: key);
+
+  @override
+  _HomeScreenState createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+
+  //Object declarations
+
+  //Offline stream handler
+  StreamSubscription? _offlineStreamSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+
+    //Simple connection checker object init
+    SimpleConnectionChecker _simpleConnectionChecker = SimpleConnectionChecker()
+      ..setLookUpAddress('pub.dev'); //Optional method to pass the lookup string
+    _offlineStreamSubscription = _simpleConnectionChecker.onConnectionChange.listen((connected) {
+      if(!connected){
+        //Launch offline screen when the device is offline
+        Navigator.of(context).pushNamedAndRemoveUntil(OfflineScreen.routeName, (route) => false);
+      }
+    });
```

Comment:
**suggestion**: This basic block of code appears several times in your screens. Code duplication is a source of bugs. This could be in provider and just expose a stream to listen for changes in each screen where needed.